### ISSUE TITLE
Support for Oculus DK2

### ISSUE THREAD

**Issue by jbohren** [ros-visualization/oculus_sdk#5]
@tfoote, one of the guys here (@zchen24) has put together a ROS package for the DK2 with SDK 0.4.3 but it's not clear what the best way is to merge these things back upstream in a clean way. For example, the current SDK would break the API. Are you interested in helping work this out?

See the fork here: https://github.com/zchen24/oculus_sdk

**Comment by jbohren** [comment]
Also @futureneer

**Comment by tfoote** [comment]
I'd suggest that we drop in the new version with DK2 support as the indigo release. We can also consider backporting and providing patches/ticketing known downstream packages in hydro depending on the size of the diff.

**Comment by futureneer** [comment]
It also may be possible to find one of the dk2 compatible api 's that are
backwards compatible with the dk1. That way we can support one api and both
devices. I will look into this.

**Comment by zchen24** [comment]
The 0.4.3 SDK should support both DK1 and DK2 with same API. 
I have some code working with DK2 but haven't tested it with DK1. 
@futureneer Do you want to try it?

**Comment by futureneer** [comment]
I can as soon as I get back to my computer with the hardware. Monday, Dec 1.

**Comment by zchen24** [comment]
Quick update: We tested SDK 0.4.3 with DK1 on 12.04 and got error message complaining missing DK1 magnetometer calibration file. See this https://forums.oculus.com/viewtopic.php?f=34&t=16261
Note: 12.04 is not officially supported. It requires libudev 1.x. But I don't think it's related.

**Comment by http404error** [comment]
The DK1 works just fine for us with SDK 0.4.3 and 0.4.4 on 14.04.
We haven't had any such success on 12.04.

**Comment by UltronDestroyer** [comment]
That's good to hear @http404error. I planned to release the oculus_sdk and oculus_rviz_plugins into Indigo, do you have any concerns that you ran into?

**Comment by http404error** [comment]
Even with the old SDK, oculus_rviz_plugins appears to function fine on indigo. I'll do some more testing, but I haven't run into anything yet.

It would be simple to upgrade to 0.4.4 for indigo via https://github.com/jherico/OculusSDK, with which oculus_rviz_plugins builds and runs just fine as well, although that unfortunately does not make the DK2  work just yet, so the benefits are marginal at best.

**Comment by UltronDestroyer** [comment]
I was able to compile oculus_sdk on 14.04 fine as well. My plan of action is to release the current oculus_sdk into Indigo and then additions (such as the DK2) can be merged and released into a future version of the package on Indigo (which can support DK1 & DK2). What do you think about that?

**Comment by http404error** [comment]
That seems reasonable. I don't know how in-demand these packages are, but I don't see any notable drawbacks to releasing now. Judging by the number of people who have failed to add DK2 support so far, it could be some time before it's ready.

**Comment by rebhuhnd** [comment]
I have been working on creating DK2 support for the Oculus SDK 0.5.0.1 as a ROS catkin package, though my approach is different than this one and it would be difficult to integrate with the existing system.

The issue I noticed seems to be that the Oculus SDK is adapted here into a cmake system,  which doesn't cleanly update when a new SDK is released. I opted to make [an SDK](https://github.com/rebhuhnd/ros_ovr_sdk) that made few modifications to the original SDK and called the makefile within the SDK during build time in order to build a shared library inside of the package.

It seems promising and forwards-compatible for when newer SDK versions are released. I thought I would add this approach to the table.

Of course, any newer versions will have broken API calls as the Oculus SDK changes, so I have decided to create branches/releases based on the version of the SDK.
